modules/idLibrary.js

```diff
@@ -0,0 +1,238 @@
+import {getGlobal} from '../src/prebidGlobal.js';
+import {ajax} from '../src/ajax.js';
+import {config} from '../src/config.js';
+import * as utils from '../src/utils.js';
+import MD5 from 'crypto-js/md5.js';
+
+let email;
+let conf;
+const LOG_PRE_FIX = 'ID-Library: ';
+const CONF_DEFAULT_OBSERVER_DEBOUNCE_MS = 250;
+const CONF_DEFAULT_FULL_BODY_SCAN = true;
+const OBSERVER_CONFIG = {
+  subtree: true,
+  attributes: true,
+  attributeOldValue: false,
+  childList: true,
+  attirbuteFilter: ['value'],
+  characterData: true,
+  characterDataOldValue: false
+};
+const logInfo = createLogInfo(LOG_PRE_FIX);
+const logError = createLogError(LOG_PRE_FIX);
+
+function createLogInfo(prefix) {
+  return function (...strings) {
+    utils.logInfo(prefix + ' ', ...strings);
+  }
+}
+
+function createLogError(prefix) {
+  return function (...strings) {
+    utils.logError(prefix + ' ', ...strings);
+  }
+}
+
+function getEmail(value) {
+  const matched = value.match(/([a-zA-Z0-9._-]+@[a-zA-Z0-9._-]+\.[a-zA-Z0-9._-]+)/gi);
+  if (!matched) {
+    return null;
+  }
+  logInfo('Email found' + matched[0]);
+  return matched[0];
+}
+
+function bodyAction(mutations, observer) {
+  logInfo('BODY observer on debounce called');
+  // If the email is found in the input element, disconnect the observer
+  if (email) {
+    observer.disconnect();
+    logInfo('Email is found, body observer disconnected');
+    return;
+  }
+
+  const body = document.body.innerHTML;
+  email = getEmail(body);
+  if (email != null) {
+    logInfo(`Email obtained from the body ${email}`);
+    observer.disconnect();
+    logInfo('Post data on email found in body');
+    postData();
+  }
+}
+
+function targetAction(mutations, observer) {
+  logInfo('Target observer called');
+  for (const mutation of mutations) {
+    for (const node of mutation.addedNodes) {
+      email = node.textContent;
+
+      if (email) {
```

Comment:
If we get an email do we need to keep going through all other mutations?

Should there only ever be one email on page?

Should we break here?

Reply:
Once email is found, data is posted and returned now